### PR TITLE
2993 update web frontend build

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Add react scripts package
- Remove custom webpack config files
- Remove old webpack dependencies
- Fix linting errors by setting to warnings
- Fix circular dependency issue in utils

---

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -152,8 +152,32 @@
     {
       "files": ["packages/web-frontend/**"],
       "rules": {
+        "camelcase": "warn",
         "import/no-named-as-default": "off",
-        "react/destructuring-assignment": "off"
+        "import/no-cycle": "warn",
+        "jsx-a11y/click-events-have-key-events": "warn",
```
Comment (tcaiger) [inline .eslintrc:158]: By default react-scripts has these lint properties set to error which crashes the build. It would be good to fix all these lint errors at some point but that is out of scope of this PR.

File: packages/web-frontend/src/components/View/utils.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ const VIEW_TYPES = {
   ...SINGLE_VALUE_COMPONENTS,
 };
 
-export const getViewWrapper = ({ type, viewType }) => {
+export function getViewWrapper({ type, viewType }) {
```
Comment (tcaiger) [inline packages/web-frontend/src/components/View/utils.js:43]: These functions were causing an error on the hot reload when changing view.js. I believe it is because of a circular dependency issue with hot reload. This fixes it because the functions get hoisted.

File: packages/web-frontend/package.json [hunk 3]
```diff
@@ -11,20 +11,22 @@
   "author": "Beyond Essential Systems <admin@tupaia.org> (https://beyondessential.com.au)",
   "main": "dist/index.js",
   "scripts": {
+    "start-new": "PORT=8088 react-scripts start",
+    "build-new": "react-scripts build",
```
Comment (edmofro) [inline packages/web-frontend/package.json:15]:
```suggestion
```

File: .eslintrc (hunk 1 above)
Comment (edmofro) [reply .eslintrc:158]: Do any of them warrant being changed in our central eslint config?

**Review by edmofro — APPROVED** [review]
Thanks for this work, very nice to have some of those things cleaned up and feel a little less behind!

File: .eslintrc (hunk 1 above)
Comment (tcaiger) [reply .eslintrc:158]: hmmm I don't think so. The other create-react-app apps seem to have the linting options as warn by default. From my lookup, this is because they are on a different version of react-scripts to the one I just installed. It may be that in the future react will enforce these lint rules more strictly so perhaps at that time we can tackle it.
